packages/malloy/src/lang/malloy-to-ast.ts

```diff
@@ -1996,4 +1996,27 @@ export class MalloyToAST
       pcx
     );
   }
+
+  visitExprWarnIn(pcx: parse.ExprWarnInContext): ast.ExprLegacyIn {
+    const expr = this.getFieldExpr(pcx.fieldExpr());
+    const isNot = !!pcx.NOT();
+    const from = pcx.fieldExprList().fieldExpr();
+    const inStmt = this.astAt(
+      new ast.ExprLegacyIn(
+        expr,
+        isNot,
+        from.map(f => this.getFieldExpr(f))
+      ),
+      pcx
+    );
+    this.warnWithReplacement(
+      'sql-in',
+      `Use = (a|b|c) instead of${isNot ? ' NOT' : ''} IN (a,b,c)`,
+      this.parseInfo.rangeFromContext(pcx),
+      `${this.getSourceCode(pcx.fieldExpr())} ${isNot ? '!=' : '='} (${from
+        .map(f => this.getSourceCode(f))
+        .join('|')})`
```

Comment:
I might format this with `.join(' | ')` to generate `x != (a | b | c)` rather than `x != (a|b|c)`, and honestly I would probably remove the parens, and just suggest `x != a | b | c`

Reply:
I was worried about "x in (a,b,c) and y = z" type of expressions turning into x = a | b | c and y = z so i left the parens in.

I went back and forth on spaces, went with "|" but have no strong attachment, you want me to change it?

Reply:
added space